presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -776,22 +788,80 @@ public Page getNextPage()
                 if (qualifyingSet == null) {
                     qualifyingSet = new QualifyingSet();
                 }
-                qualifyingSet.setRange((int) currentGroupRowCount);
+                qualifyingSet.setRange(0, Math.min(ariaBatchRows, (int) currentGroupRowCount));
```

Comment:
Instead of casting to int, consider changing the type of currentGroupRowCount to int and use Math.toIntExact to set its value.